core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -202,29 +207,12 @@ public void runInContext() {
     if (!deadlineExceeded) {
       updateTimeoutHeaders(effectiveDeadline, callOptions.getDeadline(),
           context.getDeadline(), headers);
-      ClientTransport transport = clientTransportProvider.get(
-          new PickSubchannelArgsImpl(method, headers, callOptions));
-      Context origContext = context.attach();
-      try {
-        stream = transport.newStream(method, headers, callOptions);
-      } finally {
-        context.detach(origContext);
-      }
+      stream = new RetriableStream<ReqT>(
```

Comment:
TODO: add and handle the case of `HedgingStream<ReqT>` for hedging, which will maintain a set of `Substreams`.